### DISCUSSION TITLE
Can anyone explain how fastapi and next.js cooperate under the hood?

### DISCUSSION THREAD

**Discussion by qu3vipon** [reflex-dev/reflex#457] [Q&A]
I'm struggling with deploying my pynecone docker container through AWS Elastic Beanstalk which uses Nginx for reverse proxy.

I've read [Self Hosting](https://pynecone.io/docs/hosting/self-hosting), but I can't say that I fully understand how everything works together.

I'm still getting this error:
> WebSocket connection to 'ws://0.0.0.0:8000/event' failed:

**Comment by Lendemor** [comment]
I've shared an Nginx config on the discord if you need it, but I believe the problem you are running into is linked to that issue : #441.

**Reply by qu3vipon** [reply]
Ok, I'll check it out.

I exposed fastapi on port 8000 specified in Dockerfile.

But somehow my frontend can't connect to the websocket on my backend, I couldn't find out the reason.

AWS Elastic Beanstalk doesn't support multiple port mapping, so I tried with bare EC2 but it didn't work as well.

**Reply by qu3vipon** [reply]
I ended up deploying in bare ec2 with Nginx and docker.
Your Nginx config helped a lot. Much appreciated.

### TL;DR
- Update `nginx.conf` to route /event path to port 8000. (v0.1.14)

Here's my configuration snippet.
```
http {
	# ...
	map $http_upgrade $connection_upgrade {
            default "upgrade";
	}
	
	upstream frontend {
	    server 127.0.0.1:3000;
	}
	
	upstream backend {
	    server 127.0.0.1:8000;
	}
	
	server {
	    listen 80 default_server;
	    
	    location / {
                proxy_pass http://frontend;
	        proxy_http_version 1.1;
	        
	        proxy_set_header Connection $connection_upgrade;
	        proxy_set_header Upgrade $http_upgrade;
	        proxy_set_header Host $host;
	        proxy_set_header X-Real-IP $remote_addr;
	        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
	    }
	
	    location /event {
	        proxy_pass http://backend;
	        proxy_http_version 1.1;
	        
	        proxy_set_header Connection $connection_upgrade;
	        proxy_set_header Upgrade $http_upgrade;
	        proxy_set_header Host $host;
	        proxy_set_header X-Real-IP $remote_addr;
	        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
	    }
        }
}
```
